docker/env/hydra.sh

```diff
@@ -5,7 +5,7 @@ set -eo pipefail
 CMD=$@
 DOCKER_ENV_DIR=$(readlink -f "$0")
 DOCKER_ENV_DIR=$(dirname "${DOCKER_ENV_DIR}")
-DOCKER_REPO=docker.io/scylladb/hydra
+DOCKER_REPO=scylladb/hydra
```

Comment:
One issue with this change is that it seems this repo link was added in #4648 and changing it back might break the podman support.

Reply:
So maybe we need the "docker.io" to be part of the "docker pull" command, not of DOCKER_REPO?
The command "docker images docker.io/scylladb/hydra" simply doesn't work.
I can try later to check whether my change works with podman, and if not, what does.

Reply:
`podman images` does work with the full path

I think this should work:
```diff
diff --git i/docker/env/hydra.sh w/docker/env/hydra.sh
index c6708906..86e72ff0 100755
--- i/docker/env/hydra.sh
+++ w/docker/env/hydra.sh
@@ -5,7 +5,7 @@ set -eo pipefail
 CMD=$@
 DOCKER_ENV_DIR=$(readlink -f "$0")
 DOCKER_ENV_DIR=$(dirname "${DOCKER_ENV_DIR}")
-DOCKER_REPO=docker.io/scylladb/hydra
+DOCKER_REPO=scylladb/hydra
 SCT_DIR=$(dirname "${DOCKER_ENV_DIR}")
 SCT_DIR=$(dirname "${SCT_DIR}")
 VERSION=v$(cat "${DOCKER_ENV_DIR}/version")
@@ -157,6 +157,11 @@ elif which podman >/dev/null 2>&1 ; then
 else
   die "Please make sure you install either podman or docker on this machine to run hydra"
 fi
+is_podman="$($tool --help | { grep -o podman || :; })"
+
+if [ -z "$is_podman" ]; then
+  DOCKER_REPO="docker.io/$DOCKER_REPO"
+fi
 
 if [[ ${USER} == "jenkins" || -z "`$tool images ${DOCKER_REPO}:${VERSION} -q`" ]]; then
     echo "Pull version $VERSION from Docker Hub..."
@@ -198,7 +203,6 @@ AWS_OPTIONS=$(env | sed -n 's/^\(AWS_[^=]\+\)=.*/--env \1/p')
 # export all JENKINS_* env vars into the docker run
 JENKINS_OPTIONS=$(env | sed -n 's/^\(JENKINS_[^=]\+\)=.*/--env \1/p')
 
-is_podman="$($tool --help | { grep -o podman || :; })"
 docker_common_args=()
 
 function EPHEMERAL_PORT() {

```

I wrongly assume all the docker command are working with a full path like that.

Reply:
I chose a simpler solution - we can use docker.io/.. just for the pull, and without the docker.io for all other uses - and it seems to work (I checked on both docker and podman).